### DISCUSSION TITLE
Can't turn BUILD_ZLIB off in cmake 3.29.0 (even though it should be off by default)

### DISCUSSION THREAD

**Discussion by ThibeauSW-1951556** [assimp/assimp#5539] [Q&A]
As the title implies I have pulled the repo directly and have been trying to build a static .lib file using cmake

Checking the file options at the bottom of the install instructions page I noticed that ASSIMP_BUILD_ZLIB is supposed to be off by default but when I use Cmake GUI it is in fact on

![image](https://github.com/assimp/assimp/assets/145563047/bd3c925c-04c9-4aba-ad97-2dbb49f1b30a)

now if I try to dissable it i get the following error
![image](https://github.com/assimp/assimp/assets/145563047/65c0854b-86c5-4ae6-a6e1-712d8b578863)

now since from what I understand this option needs to be off so that I don't need to install a separate instance of zlib

If anyone knows how I can fix this any help would be appreciated.

**Comment by ThatOSDev** [comment]
ZLIB comes with the assimp project. It's ON by default because of it needing to be required.

Just added info here, in case you were not aware, when you setup your project, you add -lassimp to your project, but will you also add -lz as well. No other externals needed if you are coding in C++ (maybe depending on your setup). If you code in C, like I do, then simply add the libraries in this order ...
```
-lassimp -lstdc++ -lz
```
If on windows, you add -lgdi32 BEFORE the -lassimp. In fact, any other library must be added BEFORE it.

From my experience, there is no way around the requirement for the Zlib library. 

Hope this info helps.
